sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerTest.java

```diff
@@ -451,11 +453,97 @@ public void extractContextInvalidDiagnosticId() {
         assertEquals(spanContext, invalidSpanContext);
     }
 
+    @Test
+    public void addEventWithNonNullEventName() {
+        // Arrange
+        final String eventName = "event-0";
+
+        // Act
+        openTelemetryTracer.addEvent(eventName, null, null);
+
+        // Assert
+        final ReadableSpan recordEventsSpan = (ReadableSpan) tracingContext.getData(PARENT_SPAN_KEY).get();
+        List<EventData> eventData = recordEventsSpan.toSpanData().getEvents();
+        assertNotNull(eventData);
+        assertEquals(1, eventData.size());
+        assertEquals(eventName, eventData.get(0).getName());
+    }
+
+    @Test
+    public void addEventWithAttributes() {
+        // Arrange
+        final String eventName = "event-0";
+        Map<String, Object> input = new HashMap<>() {{
+            put("attr1", "value1");
+            put("attr2", true);
+            put("attr3", 1L);
+            put("attr4", 1.0);
+            put("attr5", new double[] {1.0, 2.0, 3.0});
+            put("attr6", null);
+        }};
+
+        // Act
+        openTelemetryTracer.addEvent(eventName, input, null);
+
+        // Assert
+        final ReadableSpan recordEventsSpan = (ReadableSpan) tracingContext.getData(PARENT_SPAN_KEY).get();
+        List<EventData> eventData = recordEventsSpan.toSpanData().getEvents();
+        assertNotNull(eventData);
+        assertEquals(1, eventData.size());
+        assertEquals(eventName, eventData.get(0).getName());
+        Attributes attributes = eventData.get(0).getAttributes();
+        assertEquals(input.size() - 1, attributes.size());
+        Attributes expectedEventAttrs = Attributes.builder()
+            .put("attr5", "value1")
+            .put("attr2", true)
+            .put("attr3", 1L)
+            .put("attr4", 1.0)
+            .put("attr5", new double[] {1.0, 2.0, 3.0})
```

Comment:
There are 2 `attr5`s added to this map. The double array will overwrite the string value.